src/components/tab/SearchInput.vue

```diff
@@ -0,0 +1,299 @@
+				v-model="search"
+				:placeholder="placeholder"
+				type="text"
+				@keyup="makeSearchRequest">
+		</div>
+		<div v-if="searchResults.length>0" class="search-list">
+			<div v-for="wp in searchResults" :key="wp" class="workPackage-item">
+				<div class="wp-info">
+					<div class="wp-info__filter-wp">
+						<div class="filter-project-type-status">
+							<div class="filter-project-type-status__project">
+								{{ wp.project }}
+							</div>
+							<div class="filter-project-type-status__type" :style="{'color':wp.typeCol}">
+								{{ wp.typeTitle }}
+							</div>
+							<div class="filter-project-type-status__status" :style="{'background-color':wp.statusCol}">
+								{{ wp.statusTitle }}
+							</div>
+						</div>
+						<div class="filter-assignee">
+							<div class="filter-assignee__avatar">
+								<img class="userImage" :src="wp.picture" alt="Avatar">
+							</div>
+							<div class="filter-assignee__assignee">
+								{{ wp.assignee }}
+							</div>
+						</div>
+					</div>
+					<div class="filter-wp-subject">
+						<div class="filter-wp-subject__subject">
+							{{ wp.subject }}
+						</div>
+					</div>
+				</div>
+			</div>
+			<div class="create-new-wp">
+				{{ "+ New work package in OpenProject" }}
+			</div>
+		</div>
+		<div v-if="state === 'error'" class="title text-center">
+			{{ t('integration_openproject', "Error connecting to OpenProject") }}
+		</div>
+	</div>
+</template>
+
+<script>
+import axios from '@nextcloud/axios'
+import { generateUrl } from '@nextcloud/router'
+import { showError } from '@nextcloud/dialogs'
+
+export default {
+	name: 'SearchInput',
+	data: () => ({
+		search: null,
+		state: 'loading',
+		searchResults: [],
+	}),
+	computed: {
+		placeholder() {
+			return t('integration_openproject', 'Search for a work package to create a relation')
+		},
+	},
+	watch: {
+		search(value, oldValue) {
+			if (value.length < 3) {
+				this.searchResults.length = 0
+			}
+		},
+	},
+	methods: {
+		async makeSearchRequest(e) {
+			const url = generateUrl('/apps/integration_openproject/work_packages')
+
+			if (this.search.length >= 3) {
+				const req = {}
+				req.params = {
+					searchQuery: this.search,
+				}
+				const response = await axios.get(url, req)
+				if (Array.isArray(response.data)) {
+					this.processWorkPackages(response.data)
+					this.state = 'ok'
+				}
+				if (response.status === 400) {
+					this.state = 'no-token'
+				} else if (response.status === 401) {
+					showError(t('integration_openproject', 'Failed to get OpenProject work packages'))
+					this.state = 'error'
+				} else {
+					// there was an error in work package processing
+					// eslint-disable-next-line
+					console.debug('error')
+				}
+
+			}
+		},
+		async processWorkPackages(workPackages) {
+
+			for (let i = 0; i < workPackages.length; i++) {
+				const statusId = workPackages[i]._links.status.href
+					? workPackages[i]._links.status.href.replace(/.*\//, '')
```

Comment:
you can make a helper function for this:
```suggestion
					? this.replacesForMe(workPackages[i]._links.status.href)
```
and implement the function inside `methods` with a nice name